GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexEntry.cs

```diff
@@ -122,7 +146,89 @@ public unsafe void ParsePath()
                     }
 
                     // We unrolled the final part calculation to after the loop, to avoid having to do a 0-byte check inside the for loop
-                    this.PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, this.PathLength - currentPartStartIndex);
+                    this.BuildingProjection_PathParts[partIndex] = LazyUTF8String.FromByteArray(pathPtr + currentPartStartIndex, this.PathLength - currentPartStartIndex);
+
+                    this.NumParts++;
+                }
+            }
+
+            /// <summary>
+            /// Parses the path without using LazyUTF8Strings. It should only be called when running a background task.
+            /// </summary>
+            /// <remarks>
+            /// Code in this method has been fine-tuned for performance. None of it is shared with 
+            /// BuildingProjection_ParsePath to avoid overhead.
+            /// </remarks>
+            public unsafe void BackgroundTask_ParsePath()

```

Comment:
As discussed offline with @sanoursa , this method can use a `StringBuilder` instead of `string[]`, and we don't need to be concerned with parts (like we do when parsing for projection updates)

Reply:
Related: it would be nice to get the profiling project set up to measure this method too, and get some numbers on it

Reply:
@sanoursa that's a good idea.  I'll work on that first to get some comparison numbers between the two options.

Reply:
Actually, it looks like we already have this:

```
        /// <summary>
        /// Force the index file to be parsed to add missing paths to the modified paths database.
        /// This method should only be used to measure index parsing performance.
        /// </summary>
        void IProfilerOnlyIndexProjection.ForceAddMissingModifiedPaths(ITracer tracer)
        {
            using (FileStream indexStream = new FileStream(this.indexPath, FileMode.Open, FileAccess.ReadWrite, FileShare.Read, IndexFileStreamBufferSize))
            {
                // Not checking the FileSystemTaskResult here because this is only for profiling
                this.indexParser.AddMissingModifiedFilesAndRemoveThemFromPlaceholderList(tracer, indexStream);
            }
        }
```

I'll make sure this still works and get us some data from test runs.

Reply:
Perf numbers:

*Commit from 10/25*
Average Time 20 runs - ValidateModifiedPaths 3337.791105 ms

*StringBuilder changes (coming in 10/26) commit*
Average Time 20 runs - ValidateModifiedPaths 1012.179935 ms